core/utils/poly-look/src/visualisations/charts/bar-charts/horizontalBarChart/horizontalBarChart.js

```diff
@@ -0,0 +1,323 @@
+          if (d.group === scale.id)
+            return this._barWidth
+              ? scale.scale(d.title) +
+                  (scale.scale.bandwidth() - this._barWidth) / 2
+              : scale.scale(d.title) + scale.scale.bandwidth() / 4;
+        }
+      })
+      .attr(
+        "height",
+        this._barWidth ||
+          ((d) => {
+            for (let scale of this._yScales) {
+              if (d.group === scale.id) return scale.scale.bandwidth() / 2;
+            }
+            return this._yScales[0].scale.bandwidth() / 2;
+          })
+      )
+      .attr("fill", this._barColor)
+      .attr("class", "bar")
+      .transition()
+      .duration(750)
+      .delay((_, i) => i * 50)
+      .attr("x", 0)
+      .attr("width", (d) => this._xScale(d.value));
+  }
+
+  _addEnteringBarValues(barGroups) {
+    barGroups
+      .append("text")
+      .attr("y", (d) => {
+        for (let scale of this._yScales) {
+          if (d.group === scale.id)
+            return (
+              scale.scale(d.title) +
+              scale.scale.bandwidth() -
+              (this._groups
+                ? barTextBottomMargin
+                : (7 * barTextBottomMargin) / 3)
+            );
+        }
+      })
+      .attr("class", "bar-value")
+      .attr("text-anchor", "end")
+      .attr("x", (d) => this._xScale(d.value) - barValueMargin)
+      .text((d) => d.value)
+      .attr("fill", "transparent")
+      .style("font-size", "10px")
+      .transition()
+      .delay(1000)
+      .duration(500)
+      .attr("fill", this._barValueColor);
+  }
+
+  _alignBarText() {
+    const xScale = this._xScale;
+    const groups = this._groups;
+    const barColor = this._barColor;
+    this.chart.selectAll(".bar-group").each(function () {
+      const node = d3.select(this);
+      const value = node.select(".bar-value");
+      const label = node.select(".bar-label");
+      const labelWidth = groups ? label.node().getBBox().width : 0;
+      if (
+        value.node().getBBox().width + labelWidth + barValueMargin >
+        xScale(+value.node().textContent)
+      ) {
+        if (groups)
+          label
+            .attr("text-anchor", "start")
+            .attr("x", (d) => barValueMargin + xScale(d.value))
+            .attr("fill", "transparent")
+            .transition()
+            .delay(1000)
+            .duration(500)
+            .attr("fill", barColor);
+        value
+          .attr("text-anchor", "start")
+          .attr(
+            "x",
+            (d) =>
+              barValueMargin + xScale(d.value) + barValueMargin + labelWidth
+          )
+          .attr("fill", "transparent")
+          .transition()
+          .delay(1000)
+          .duration(500)
+          .attr("fill", barColor);
+      }
+    });
+  }
+
+  _updateExistingBarValues(barValues) {
+    barValues
+      .attr("fill", "transparent")
+      .attr("x", 0)
+      .text((d) => d.value)
+      .raise()
+      .transition()
+      .delay(1500)
```

Comment:
It's 1500 here why?

Reply:
I think it's just how the animation works, so pretty much eye balled values.

Reply:
But is there going to be some kind of uniform animation throughout all charts? So this might be something like a "semi-slow delay" or something like that?